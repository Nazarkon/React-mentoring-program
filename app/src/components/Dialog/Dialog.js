import React from 'react';
import PropTypes from 'prop-types';
import { Portal } from 'react-portal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import './Dialog.scss';

const Dialog = ({ title, children, handleClose }) => {
  return (
    <Portal>
      <div className="dialog-container">
        <FontAwesomeIcon
          onClick={handleClose}
          className="dialog-close-icon"
          data-testid="close-icon"
          icon={faTimes}
          style={{ color: 'white' }}
        />
        <div className="dialog-container-main">
          <div className="dialog-title">{title}</div>
          <div className="dialog-custom">{children}</div>
        </div>
      </div>
    </Portal>
  );
};

Dialog.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  handleClose: PropTypes.func.isRequired
};

export default Dialog;
