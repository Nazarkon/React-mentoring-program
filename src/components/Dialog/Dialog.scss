.dialog-container{
    display: inline-block;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background: #232323;
    .dialog-close-icon{
        position: absolute;
        top: 10px;
        cursor: pointer;
        right: 10px;
        color: white;
        width: 30px;
        height: 30px;
    }
    .dialog-container-main{
        padding: 60px 50px;
    }
    .dialog-title{
        color: #FFFFFF;
        font-size: 40px;
        font-family: 'Montserrat';
        font-style: normal;
        font-weight: 300;
        margin-bottom: 38px;
    }
    .dialog-message{
        color: #FFFFFF;
        font-style: normal;
        font-family: 'Montserrat';
        font-weight: 400;
        font-size: 20px;
        line-height: 24px;
        text-align: center;
    }
    .dialog-buttons{
        text-align: right;
        margin-top: 52px;
        .dialog-button{
            display: inline-block;
            color:#FFFFFF;
            text-transform: uppercase;
            padding: 17px 57px;
            border: 1px solid transparent;
            background-color: #F65261;
                &:hover{
                    background-color: transparent;
                    color: #F65261;
                    border: 1px solid #F65261;
                }
        }
    }

}