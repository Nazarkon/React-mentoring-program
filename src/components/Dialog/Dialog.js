import React, {useState} from "react";
import { Portal } from "react-portal";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import "./Dialog.scss"

const Dialog = ({ title, children, handleClose}) => {
  return (
        <Portal>
          <div className="dialog-container">
            <FontAwesomeIcon onClick={handleClose} className="dialog-close-icon" data-testid="close-icon" icon={faTimes}  style={{ color: 'white' }} />
              <div className="dialog-container-main">
                <div className="dialog-title">{title}</div>
                  <div className="dialog-custom">
                    {children}
                </div>
              </div>
          </div>
        </Portal>
  );
}

export default Dialog