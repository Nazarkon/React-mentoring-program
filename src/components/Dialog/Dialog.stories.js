import React, { useState } from "react";

import Dialog from "./Dialog";
import MovieForm from "../MovieForm/MovieForm.js";
import DialogMessage from "../DialogMessages/DialogMessage";

export default {
    title: 'Components/Dialog',
    component: Dialog,
    argTypes: {
        title: {
            control: "text",
            description: "Title of a dialog window"
        },
        handleClick: {
            action: "clicked",
        },
    },
}

const mockedMovieData = {
    title: 'Pulp Fiction',
    releaseDate: '2022-01-01',
    movieURL: 'https://movie-url/example',
    rating: '8.9',
    runtime: '2h 34min', 
    overview: 'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra', 
    genre: 'comedy',
}

const TemplateAddMovie = (args) => {
    const [isOpen, setIsOpen] = useState(true)

    const handleSubmit = (data) => {
        console.log(data)
        setIsOpen(!isOpen)
    }
    return (
        <>
        {isOpen && (
            <Dialog  
                title="Add Movie" 
                isOpen={isOpen} 
                handleClose={() => setIsOpen(!isOpen)} 
            >
                <MovieForm  handleSubmit={handleSubmit}/>
            </Dialog>)}
    </>
    );
}
const TemplateEditMovie = (args) => {
    const [isOpen, setIsOpen] = useState(true);

    const handleSubmit = (data) => {
        console.log(data)
        setIsOpen(!isOpen)
    }

    return (
        <>
        { isOpen && (
        <Dialog  
            title="Edit Movie" 
            handleClose={() => setIsOpen(!isOpen)} 
        >
            <MovieForm  handleSubmit={handleSubmit} movieInfo={mockedMovieData}/>
        </Dialog>
        )}
        </>
    )
}
const TemplateDeleteMovie = (args) => {
    const [isOpen, setIsOpen] = useState(true)

    const handleSubmit = () => {
        setIsOpen(!isOpen)
    }

    return (
        <>
        {
            isOpen && (
                <Dialog
                title="DELETE MOVIE" 
                handleClose={() => setIsOpen(!isOpen)}
                >
                    <DialogMessage 
                        message={"Are you sure you want to delete this movie?"}
                        buttonText={'Confirm'}
                        handleConfirm={handleSubmit}
                    />
                </Dialog>
            )
        }
        </>
    )
}

export const AddMovie = TemplateAddMovie.bind({});
export const EditMovie = TemplateEditMovie.bind({});
export const DeleteMovie = TemplateDeleteMovie.bind({});