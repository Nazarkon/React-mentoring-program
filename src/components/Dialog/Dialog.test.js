/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom'

import Dialog from "./Dialog";

describe("Dialog component", () => {

  const title = 'Test Title';
  const body = 'Test Body';
  const handleClose = jest.fn();

    it("matches snapshot", () => {
        const { container } = render(
            <Dialog title={title} body={body} handleClose={handleClose} isOpen={true} />);
        expect(container).toMatchSnapshot();
      });
      it('Should render the dialog with the correct title and body', () => {
        const { getByText } = render(<Dialog title={title} body={body} handleClose={handleClose} isOpen={true} />);
    
        expect(getByText(title)).toBeInTheDocument();
        expect(getByText(body)).toBeInTheDocument();
      });
    
      it('Should call the handleClose function when the close icon is clicked', () => {
        const {getByTestId} = render(<Dialog title={title} body={body} handleClose={handleClose} isOpen={true} />);
    
        const closeIcon = getByTestId('close-icon');
        userEvent.click(closeIcon);
    
        expect(handleClose).toHaveBeenCalled();
      });
});