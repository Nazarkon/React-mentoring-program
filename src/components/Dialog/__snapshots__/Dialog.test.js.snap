// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component matches snapshot 1`] = `<div />`;
